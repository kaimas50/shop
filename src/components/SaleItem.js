import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import styled from 'styled-components';

const SaleItem = (props) => (
  <ItemDiv>
    <ExpandingPaper>
      {props.num}
    </ExpandingPaper>
  </ItemDiv>
);
SaleItem.propTypes = {
  num: PropTypes.number.isRequired,
};

const ItemDiv = styled.div`
  && {
    height: 150px;
    padding: 4px;

    @media only screen and (max-width: 600px) {
      width: 100%;
    }
    @media only screen and (min-width: 600px) {
      width: 33.33333%;
    }
    @media only screen and (min-width: 648px) {
      width: 25%;
    }
    @media only screen and (min-width: 820px) {
      width: 20%;
    }
    @media only screen and (min-width: 972px) {
      width: 16.66667%;
    }
    @media only screen and (min-width: 1024px) {
      max-width: 200px;
    }
  }
`;

const ExpandingPaper = styled(Paper)`
  width: 100%;
  height: 100%;
`;

export default SaleItem;
