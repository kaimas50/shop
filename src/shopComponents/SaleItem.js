import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import styled from 'styled-components';
import StepNumberInput from '../components/StepNumberInput';

const SaleItem = (props) => (
  <ItemDiv id={props.id}>
    <ExpandingPaper>
      <div>{props.name}</div>
      <img src={props.imgSrc} alt="" />
      <StepNumberInput min={0} value={props.value} onChange={props.onChange} />
      <div>{props.desc}</div>
    </ExpandingPaper>
  </ItemDiv>
);
SaleItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired,
};

const ItemDiv = styled.div`
  && {
    height: 280px;
    padding: 4px;

    @media only screen and (max-width: 600px) {
      width: 100%;
    }
    @media only screen and (min-width: 600px) {
      width: 33.33333%;
    }
    @media only screen and (min-width: 648px) {
      width: 25%;
    }
    @media only screen and (min-width: 820px) {
      width: 20%;
    }
    @media only screen and (min-width: 972px) {
      width: 16.66667%;
    }
    @media only screen and (min-width: 1024px) {
      max-width: 200px;
    }
  }
`;

const ExpandingPaper = styled(Paper)`
  width: 100%;
  height: 100%;
  padding: 4px;
`;

export default SaleItem;
