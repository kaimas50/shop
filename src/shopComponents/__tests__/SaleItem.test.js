import React from 'react';
import renderer from 'react-test-renderer';
import SaleItem from '../SaleItem';
import canFish from './can-fish.jpg';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SaleItem
        id="1"
        name="Sale Item"
        desc="Description"
        imgSrc={canFish}
        price="50"
        onChange={() => null}
        value={4}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
