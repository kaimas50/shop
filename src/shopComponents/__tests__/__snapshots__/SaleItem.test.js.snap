// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bxivhb dVuENo"
  id="1"
>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-ifAKCX lcPtQz"
  >
    <div>
      Sale Item
    </div>
    <img
      alt=""
      src="can-fish.jpg"
    />
    <div>
      50â‚¬
    </div>
    <div
      className="sc-htpNat eeSbyC"
    >
      <button
        className="MuiButtonBase-root-34 MuiIconButton-root-28 sc-bwzfXH buaYIJ"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-33"
        >
          -
        </span>
        <span
          className="MuiTouchRipple-root-37"
        />
      </button>
      <div
        className="MuiInput-root-44 MuiInput-underline-48 sc-bdVaJa jTIBAz"
      >
        <input
          aria-invalid={undefined}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-52"
          defaultValue={undefined}
          disabled={undefined}
          id={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value={4}
        />
      </div>
      <button
        className="MuiButtonBase-root-34 MuiIconButton-root-28 sc-bwzfXH buaYIJ"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-33"
        >
          +
        </span>
        <span
          className="MuiTouchRipple-root-37"
        />
      </button>
    </div>
    <div>
      Description
    </div>
  </div>
</div>
`;
